src/client/executor.rs

```diff
@@ -30,75 +32,129 @@ lazy_static! {
 }
 
 impl Client {
-    /// Executes an operation and returns the connection used to do so along with the result of the
-    /// operation. This will be used primarily for the opening of exhaust cursors.
-    #[allow(dead_code)]
-    pub(crate) async fn execute_exhaust_operation<T: Operation>(
-        &self,
-        op: &T,
-    ) -> Result<(T::O, Connection)> {
-        let server = self.select_server(op.selection_criteria()).await?;
-        let mut conn = server.checkout_connection().await?;
-        self.execute_operation_on_connection(op, &mut conn)
+    /// Execute the given operation.
+    ///
+    /// Server selection will performed using the criteria specified on the operation, if any, and
+    /// an implicit session will be created if the operation and write concern are compatible with
+    /// sessions.
+    pub(crate) async fn execute_operation<T: Operation>(&self, op: T) -> Result<T::O> {
+        let mut implicit_session = self.start_implicit_session(&op).await?;
+        self.select_server_and_execute_operation(op, implicit_session.as_mut())
             .await
-            .map(|r| (r, conn))
     }
 
-    pub(crate) async fn execute_operation_owned<T: Operation>(self, op: T) -> Result<T::O> {
-        self.execute_operation(&op, None).await
+    /// Execute the given operation with an implicit session, returning an error if one cannot be
+    /// created or the topology doesn't support them. The implicit session will be returned from
+    /// this method.
+    ///
+    /// Server selection be will performed using the criteria specified on the operation, if any.
+    pub(crate) async fn execute_operation_with_implicit_session<T: Operation>(
```

Comment:
This comment is out of date--sometimes we can't execute with an implicit session (e.g. a standalone)

Reply:
Sorry, I'm not quite sure I understand; has the comment been updated now, or does it still need to be changed?

Reply:
This GitHub comment (and the docstring comment) are both out of date. The docstring has since been updated in later commits. I now realize that "this comment" and "outdated" both could've meant several different things here lol.